src/Symfony/Component/Translation/Bridge/Lokalise/Provider/LokaliseProvider.php

```diff
@@ -0,0 +1,220 @@
+    }
+
+    /**
+     * Lokalise API recommends sending payload in chunks of up to 500 keys per request.
+     *
+     * @see https://app.lokalise.com/api2docs/curl/#transition-create-keys-post
+     */
+    private function createKeysWithTranslations(TranslatorBag $translatorBag): void
+    {
+        $keys = [];
+        $catalogue = $translatorBag->getCatalogue($this->defaultLocale);
+
+        if (!$catalogue) {
+            $catalogue = $translatorBag->getCatalogues()[0];
+        }
+
+        foreach ($translatorBag->getDomains() as $domain) {
+            foreach ($catalogue->all($domain) as $key => $message) {
+                $keys[] = [
+                    'key_name' => $key,
+                    'platforms' => ['web'],
+                    'filenames' => [
+                        'web' => $this->generateLokaliseFilenameFromDomain($domain),
+                        // There is a bug in Lokalise with "Per platform key names" option enabled,
+                        // we need to provide a filename for all platforms.
+                        'ios' => null,
+                        'android' => null,
+                        'other' => null,
+                    ],
+                    'translations' => array_map(function ($catalogue) use ($key, $domain) {
+                        return [
+                            'language_iso' => $catalogue->getLocale(),
+                            'translation' => $catalogue->get($key, $domain),
+                        ];
+                    }, $translatorBag->getCatalogues()),
+                ];
+            }
+        }
+
+        $chunks = array_chunk($keys, 500);
+
+        foreach ($chunks as $chunk) {
+            $response = $this->client->request('POST', sprintf('/projects/%s/keys', $this->projectId), [
+                'json' => ['keys' => $chunk],
+            ]);
+
+            if (200 !== $response->getStatusCode()) {
+                throw new ProviderException(sprintf('Unable to add keys and translations to Lokalise: "%s".', $response->getContent(false)), $response);
+            }
+        }
+    }
+
+    /**
+     * @see https://app.lokalise.com/api2docs/curl/#transition-download-files-post
+     */
+    private function exportFiles(array $locales, array $domains): array
+    {
+        $response = $this->client->request('POST', sprintf('/projects/%s/files/export', $this->projectId), [
+            'json' => [
+                'format' => 'symfony_xliff',
+                'original_filenames' => true,
+                'directory_prefix' => '%LANG_ISO%',
+                'filter_langs' => array_values($locales),
+                'filter_filenames' => array_map([$this, 'generateLokaliseFilenameFromDomain'], $domains),
+            ],
+        ]);
+
+        $responseContent = $response->getContent(false);
+
+        if (406 === $response->getStatusCode()
+            && 'No keys found with specified filenames.' === json_decode($responseContent, true)['error']['message']) {
```

Comment:
`) {` should be on its own line

please also replace all direct calls to `json_decode()` by `$response->toArray()` when possible in the PR